### ISSUE TITLE
Volume attachment resource creation completes before volume is attached

### ISSUE THREAD

**Issue by devjaniray** [terraform-provider-openstack/terraform-provider-openstack#277]
### Terraform Version
$ terraform -v
Terraform v0.11.6-dev
+ provider.null v1.0.0
+ provider.openstack (unversioned)
+ provider.template v1.0.0

Originally reproduced with:
$ terraform -v
Terraform v0.11.3
+ provider.null v1.0.0
+ provider.openstack v1.2.0
+ provider.template v1.0.0

### Affected Resource(s)
Please list the resources as a list, for example:
- resource "openstack_compute_volume_attach_v2"

This issue does not affect multiple resources or Terraform core.

### Terraform Configuration Files
```hcl
resource "openstack_blockstorage_volume_v2" "tf_db_volume" {
  name        = "${var.instance_name}_tf_db_volume"
  size        = "${var.db_volume_size}"
  snapshot_id = "${var.db_volume_snapshot_id}"
}

resource "openstack_compute_volume_attach_v2" "tf_db_volume_attachment" {
  volume_id   = "${openstack_blockstorage_volume_v2.tf_db_volume.id}"
  instance_id = "${openstack_compute_instance_v2.tf_instance.id}"
}

...

resource "null_resource" "deploy_db" {
  # The following resources must be ready before deploying the database
  depends_on = [
    "openstack_compute_volume_attach_v2.tf_db_volume_attachment",
    "openstack_compute_instance_v2.tf_instance",
  ]
...
}

```

### Debug Output
N/A

### Panic Output
N/A

### Expected Behavior
The resource creation is complete only after the volume status is "in-use" (or 'reserve'?) Or provide a way to determine the volume status on completion.

### Actual Behavior
The resource creation completes while the volume is still attaching.

### Steps to Reproduce
Please list the steps required to reproduce the issue, for example:
1. `terraform apply`
2. Check volume status in another terminal by running `openstack volume list`
'terraform apply' output shows that the volume attachment resource is complete, while 'openstack volume list' shows the volume status as 'reserved' or 'attaching'.

### Important Factoids
This behavior was seen with OpenStack Pike and Queens. 

### References
No known GitHub issues. However this issue was discussed in the Terraform Google Group topic: https://groups.google.com/forum/#!topic/terraform-tool/48A1A3qJLaA

**Comment by fgimenezm** [comment]
A side effect of this behaviour is that if the actual attachment fails in the backend after terraform has already flagged the resource "complete" then the failure will go undetected and terraform will finish successfully even though the attachment isn't really there.

**Comment by jichenjc** [comment]
so we need update to query the status 
https://docs.openstack.org/api-ref/block-storage/v2/index.html?expanded=show-volume-details-detail#show-volume-details

and by the way, we should use v3 API as v2 API is deprecated now

**Comment by fgimenezm** [comment]
I think this is fixed by #1106
